portfolio/src/main/webapp/script.js

```diff
@@ -61,3 +61,171 @@ function addRandomFunFact() {
+    const numPages = document.querySelectorAll('#page-list li').length;
+    if (currentPage.firstElementChild.innerText == numPages - 2) { return; }
+
+    currentPage.nextSibling.classList.add('active');
+  } else {
+    event.currentTarget.classList.add('active');
+  }
+
+  currentPage.classList.remove('active');
+
+  // Fetches comments on that page without recreating pagination.
+  fetchComments(false);
+}
+
+/* 
+ * Fetches the comments data on the current page from the server and displays them.
+ */
+function fetchComments(createNewPagination = false) {
+  const currentPage = document.querySelector('.page-item.active');
+  var currentPageNumber = currentPage.firstElementChild.innerText;
+
+  const selectElement = document.getElementById('num-comments-per-page');
+  const commentsPerPage = selectElement.options[selectElement.selectedIndex].value;
+
+  fetch('/data').then(response => response.json()).then(comments => {
+    const commentsListElement = document.getElementById('comments-list');
+    commentsListElement.innerHTML = '';
+
+    if (createNewPagination) {
+      currentPageNumber = createPagination(Math.ceil(comments.length / commentsPerPage), currentPageNumber);
+    }
+
+    const startIndex = (currentPageNumber - 1) * commentsPerPage;
+    const endIndex = Math.min(Number(commentsPerPage) + Number(startIndex), comments.length);
+    for (var i = startIndex; i < endIndex; i++) {
+      commentsListElement.appendChild(createCommentElement(comments[i]));
+    }
+  });
+}
+
+/* 
+ * Creates an <li> element containing a comment.
+ */
+function createCommentElement(comment) {
+  const commentElement = document.createElement('li');
+  commentElement.className = 'comment';
+
+  const divElement = document.createElement('div');
+
+  const nameElement = document.createElement('strong');
+  nameElement.className = 'text-success';
+  nameElement.innerText = '@' + comment.name + ' - ' + comment.mood;
+  const timeElement = document.createElement('span');
+  timeElement.className = 'pull-right text-muted';  // Adds Bootstrap classes to style the timeElement.
+  timeElement.innerText = convertToDateTime(comment.timestamp);
+  const contentElement = document.createElement('p');
+  contentElement.innerText = comment.content;
+  
+  divElement.appendChild(timeElement);
+  divElement.appendChild(nameElement);
+  divElement.appendChild(contentElement);
+
+  const deleteButtonElement = document.createElement('button');
+  deleteButtonElement.innerText = 'Delete';
+  deleteButtonElement.addEventListener('click', () => {
+    deleteComment(comment);
+    commentElement.remove();    // Removes the comment from the DOM.
+  });
+
+  commentElement.appendChild(divElement);
+  commentElement.appendChild(deleteButtonElement);
+  return commentElement;
+}
+
+/* 
+ * Converts a timestamp in milliseconds to a formatted date/time string.
+ * Returns dd/mm/yyyy if not on same day; otherwise, returns number of hours ago. 
```

Comment:
update JSDoc, and also "Returns date with format dd/mm/yyyy ...."

Reply:
Got it